src/components/SourceEditForm/SourceEditModal.js

```diff
@@ -109,7 +109,7 @@ const SourceEditModal = () => {
                                 isHorizontal: true
                             }}
                             canReset
-                            disableSubmit={['submitting']}
+                            disableSubmit={['submitting', 'pristine']}
```

Comment:
is there any async validation? We might want to add `validating` to the list.

Reply:
Validating is disabling button by default in the common formTemplate

```jsx
disabled={submitting || validating || disableSubmit}
```